Common/include/CConfig.hpp

```diff
@@ -8171,6 +8178,72 @@ class CConfig {
    */
   su2double GetFFD_Tol(void) const { return FFD_Tol; }
 
+  /*!
+   * \brief Get information about whether to do a check on self-intersections within the FFD box based on value of the Jacobian determinant.
+   * \return <code>TRUE</code> if FFD intersection prevention is active; otherwise <code>FALSE</code>.
+   */
+  bool GetFFD_IntPrev(void) const { return FFD_IntPrev; }
+
+  /*!
+   * \brief Get the maximum number of iterations in the recursive convexity check procedure in deformation.
+   * \return Maximum number of iterations.
+   */
+  unsigned short GetFFD_IntPrev_Iter(void) const { return FFD_IntPrev_Iter; }
+
+  /*!
+   * \brief Get the maximum recursion depth in the recursive convexity check procedure in deformation.
+   * \return Maximum recursion depth.
+   */
+  unsigned short GetFFD_IntPrev_Depth(void) const { return FFD_IntPrev_Depth; }
+
+  /*!
+   * \brief Get information about whether to do a check on convexity of the mesh elements.
+   * \return <code>TRUE</code> if convexity check is active; otherwise <code>FALSE</code>.
+   */
+  bool GetConvexity_Check(void) const { return ConvexityCheck; }
+
+  /*!
+   * \brief Get the maximum number of iterations in the recursive convexity check procedure in deformation.
+   * \return Maximum number of iterations.
+   */
+  unsigned short GetConvexityCheck_Iter(void) const { return ConvexityCheck_Iter; }
+
+  /*!
+   * \brief Get the maximum recursion depth in the recursive convexity check procedure in deformation.
+   * \return Maximum recursion depth.
+   */
+  unsigned short GetConvexityCheck_Depth(void) const { return ConvexityCheck_Depth; }
+
+  /*!
+   * \brief Set the initial values for the total deformation of the design variables.
+   * \param[in] totalDef - total deformation of all design variables
+   */
+  void Initialize_TotalDeformation(su2double** totalDef) { TotalDeformation = totalDef; }
+
+  /*!
+   * \brief Set the total deformation value of each design variable.
+   * \param[in] val_dv - design variable
+   * \param[in] val_ind - index of design variable
+   * \param[in] val - value of deformation magnitude
+   */
+  void SetTotalDeformation(unsigned short val_dv, unsigned short val_ind, su2double val) {TotalDeformation[val_dv][val_ind] = val; }
+
+  /*!
+   * \brief Add a value to the total deformation value of each design variable.
+   * \param[in] val_dv - design variable
+   * \param[in] val_ind - index of design variable
+   * \param[in] val - value of deformation magnitude that is added
+   */
+  void AddTotalDeformation(unsigned short val_dv, unsigned short val_ind, su2double val) {TotalDeformation[val_dv][val_ind] += val; }
```

Comment:
Can this "total deformation" logic be moved out of CConfig? I and other folks think the config is already responsible for too much.

Reply:
Is there a place outside of cconfig that makes sense? If this place does not exist I think we should make this point into a separate issue/PR, since it goes beyond this PR. How do we separate these functions into logical groups? Let's discuss what kind of splitting makes sense.

Reply:
It is just a variable that is used in the deformation routines. It can be passed as an argument when needed. This does not have to be complicated.
The purpose of CConfig is to read user inputs, using it as this "global" object with mutable state is not good.

Reply:
OK, clear, and I agree.